native/src/routes/OffersContainer.tsx

```diff
@@ -37,83 +23,51 @@ type OffersContainerProps = {
 const OffersContainer = ({ navigation, route }: OffersContainerProps): ReactElement => {
   const showSnackbar = useSnackbar()
   const { cityCode, languageCode } = useCityAppContext()
+  const { data, ...response } = useLoadOffers({ cityCode, languageCode })
   const { navigateTo } = useNavigate()
-  const cities = useCities()
-  const { t } = useTranslation('offers')
 
-  const routeInformation = { route: OFFERS_ROUTE, languageCode, cityCode }
-  useSetShareUrl({ navigation, routeInformation, route })
+  const availableLanguages = data?.languages.map(it => it.code)
+  const shareUrl = urlFromRouteInformation({ route: OFFERS_ROUTE, languageCode, cityCode })
+  useHeader({ navigation, route, availableLanguages, data, shareUrl })
 
-  const request = useCallback(async () => {
-    const apiUrl = await determineApiUrl()
-    return createOffersEndpoint(apiUrl).request({
-      city: cityCode,
+  const navigateToOffer = (tile: TileModel) => {
+    const { title, path, isExternalUrl, postData } = tile
+    if (isExternalUrl && postData) {
+      // HTTP POST is neither supported by the InAppBrowser nor by Linking, therefore we have to open it in a webview
+      navigation.push(EXTERNAL_OFFER_ROUTE, {
+        url: path,
+        shareUrl: path,
+        postData,
+      })
+    } else if (isExternalUrl) {
+      openExternalUrl(path).catch(error => showSnackbar(error.message))
+    } else if (data?.offers.find(offer => offer.title === title)?.alias === SPRUNGBRETT_OFFER_ROUTE) {
```

Comment:
What happens if none of these three if-cases are true? Should an else case which reports an error be added?

Reply:
Right now not possible since the only internal extra on native is the sprungbrett offer and `isExternalUrl` is set to `      isExternalUrl: offer.alias !== SPRUNGBRETT_OFFER_ROUTE,
`